agent/api/task.go

```diff
@@ -190,6 +192,13 @@ func (task *Task) DockerConfig(container *Container) (*docker.Config, *DockerCli
 }
 
 func (task *Task) dockerConfig(container *Container) (*docker.Config, *DockerClientConfigError) {
+	image := container.Image
+	if strings.HasPrefix(container.Image, "s3://") {
+		hash := md5.New()
+		tag := hex.EncodeToString(hash.Sum([]byte(container.Image)))
+		image = task.Arn + ":" + tag
```

Comment:
@tlunter It does have colons, but according to the [v1 spec](https://github.com/docker/docker/blob/master/image/spec/v1.md#terminology), that's okay (search for "Repository"). Arn format: `arn:aws:ecs:$region:$account_id:task-definition/$task_name:$task_revision`